Sprint 15 - ShareIt v.1

DmitryBrosalin opened the pull request:
Подключена база данных PostgreSQL.
Добавлена возможность бронирования и создания комментариев после бронирования.

DmitryBrosalin committed:
Sprint 15 - ShareIt v.1
Подключена база данных PostgreSQL.
Добавлена возможность бронирования и создания комментариев после бронирования.

DmitryBrosalin committed:
Sprint 15 - ShareIt v.1
Подключена база данных PostgreSQL.
Добавлена возможность бронирования и создания комментариев после бронирования.

DmitryBrosalin committed:
Sprint 15 - ShareIt v.1
Подключена база данных PostgreSQL.
Добавлена возможность бронирования и создания комментариев после бронирования.

DmitryBrosalin committed:
Sprint 15 - ShareIt v.1
Подключена база данных PostgreSQL.
Добавлена возможность бронирования и создания комментариев после бронирования.

slavrikk commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Query("SELECT b FROM Booking b JOIN FETCH b.item WHERE item.owner.id = ?1 AND " +
+            "b.start < CURRENT_TIMESTAMP AND b.end > CURRENT_TIMESTAMP AND " +
+            "b.status LIKE 'APPROVED' ORDER BY b.end DESC")
```
Лучше `b.status= ?2` и передать статус в метод

slavrikk commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Query("SELECT b FROM Booking b JOIN FETCH b.item WHERE item.owner.id = ?1 AND " +
+            "b.end < CURRENT_TIMESTAMP AND " +
+            "b.status LIKE 'APPROVED' ORDER BY b.end DESC")
```
Аналогично

slavrikk commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Query("SELECT b FROM Booking b JOIN FETCH b.item WHERE item.owner.id = ?1 AND " +
+            "b.start > CURRENT_TIMESTAMP AND " +
+            "b.status LIKE 'APPROVED' ORDER BY b.end DESC")
```
Аналогично

slavrikk commented on src/main/java/ru/practicum/shareit/booking/BookingService.java:
```diff
@@ -0,0 +1,127 @@
+            throw new NotFoundException("Пользователь не найден.");
+        }
+        state = state.toUpperCase();
+        return switch (state) {
```
Лучше state преобразовать в enum

slavrikk commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDtoFromController.java:
```diff
@@ -0,0 +1,14 @@
+
+@AllArgsConstructor
+@Data
+public class BookingDtoFromController {
```
Лучше переименовать как `BookingDtoRequest`

slavrikk commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDtoToController.java:
```diff
@@ -0,0 +1,20 @@
+
+@Data
+@AllArgsConstructor
+public class BookingDtoToController {
```
Лучше переименовать как `BookingDtoResponse`

slavrikk commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,35 @@
+  name VARCHAR(255) NOT NULL,
+  description VARCHAR(255) NOT NULL,
+  available BOOLEAN NOT NULL,
+  user_id BIGINT NOT NULL,
```
Не указан foreign key на таблицу users

slavrikk commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,35 @@
+CREATE TABLE IF NOT EXISTS bookings (
+    id BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL,
+    item_id BIGINT NOT NULL,
+    booker_id BIGINT NOT NULL,
```
Не указан foreign key на таблицу users

slavrikk commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,35 @@
+
+CREATE TABLE IF NOT EXISTS comments (
+    id BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL,
+    item_id BIGINT NOT NULL,
```
Не указан foreign key на таблицу items

DmitryBrosalin committed:
Sprint 15 - ShareIt v.2
Добавлены foreign keys в схему создания таблиц.
Изменены названия BookingDto-классов.
В запросы на получение бронирований к БД передаются enum состояния бронирований.
